### PR TITLE
update to ochami-deployment compose file

### PR DESCRIPTION
**Pull request by travisbcotton**
Added lines to add the ochami.yaml file as a docker secret. 
Requires an update to ochami-init to work: https://github.com/OpenCHAMI/ochami-init/pull/4

### CODE REVIEW

**Comment by alexlovelltroy** [comment]
Added the not ready label to indicate that we should hold off on merging until we also have a new release version from ochami-init.

File: lanl/docker-compose/generate-configs.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Ochami config file location
+OCHAMI_CONFIG=/etc/ochami/ochami-config.yaml
+
+#Set DB passwords 
+echo "POSTGRES_PASSWORD=$(openssl rand -base64 32)" > .env
+echo "BSS_POSTGRES_PASSWORD=$(openssl rand -base64 32)" >> .env
+echo "OCHAMI_CONFIG=$OCHAMI_CONFIG" >> .env
+
+#Copy Config template to real config location
+mkdir -p /etc/ochami
```
Comment (synackd) [inline lanl/docker-compose/generate-configs.sh:12]: I'm a bit worried about this running (and erring if the user does not have privileges) if the user changes `OCHAMI_CONFIG`. Perhaps you could have:

```bash
OCHAMI_DIR=${OCHAMI_DIR:-/etc/ochami}
OCHAMI_CONFIG=${OCHAMI_CONFIG:-${OCHAMI_DIR}/ochami-config.yaml}
```

and then do

```bash
mkdir -p ${OCHAMI_DIR}
```

instead.

The `${<varname>:-<text>}` is substituted with `<text>` if the environment variable `<varname>` is not set. That way, a user could override those variables.

All this if you would rather not use positional arguments or flags. 😉

Comment (travisbcotton) [reply lanl/docker-compose/generate-configs.sh]: Great suggestion. added that in

Comment (synackd) [reply lanl/docker-compose/generate-configs.sh]: Looks good!

**Comment by synackd** [comment]
Hmm... During testing, it seems inconvenient to have to set two separate variables to change where to put the config. It seems better to, e.g:

```
$ OCHAMI_CONFIG=/path/to/config/ochami.yaml ./generate-configs.sh
```

and the script will `mkdir -p /path/to/config`, rather than to have to:

```
$ OCHAMI_DIR=/path/to/config OCHAMI_CONFIG=${OCHAMI_DIR}/ochami.yaml ./generate-configs.sh
```

Here is the change to be able to do the first:

```diff
diff --git a/lanl/docker-compose/generate-configs.sh b/lanl/docker-compose/generate-configs.sh
index a1496a8..6d5cef5 100644
--- a/lanl/docker-compose/generate-configs.sh
+++ b/lanl/docker-compose/generate-configs.sh
@@ -1,8 +1,7 @@
 #!/bin/bash

 # Ochami config file location
-OCHAMI_DIR=${OCHAMI_DIR:-/etc/ochami}
-OCHAMI_CONFIG=${OCHAMI_CONFIG:-${OCHAMI_DIR}/ochami-config.yaml}
+OCHAMI_CONFIG=${OCHAMI_CONFIG:-/etc/ochami/ochami-config.yaml}

 if [ -f $OCHAMI_CONFIG ]
 then
@@ -16,7 +15,7 @@ echo "BSS_POSTGRES_PASSWORD=$(openssl rand -base64 32)" >> .env
 echo "OCHAMI_CONFIG=$OCHAMI_CONFIG" >> .env

 #Copy Config template to real config location
-mkdir -p $OCHAMI_DIR
+mkdir -p $(dirname OCHAMI_CONFIG)
 cp configs/ochami-template.yaml $OCHAMI_CONFIG

 #replace passwords in template.
```

**Review by alexlovelltroy — DISMISSED** [review]
LGTM

**Comment by synackd** [comment]
Looks good. One last thing: Could the script be made executable?

**Review by alexlovelltroy — DISMISSED** [review]
LGTM

**Pull request closed without merging**